### PR TITLE
Isolating makeImage in a domain

### PR DESCRIPTION
**Pull request by ElNounch**

### CODE REVIEW

File: badge.js [hunk 1]
```diff
@@ -96,4 +97,13 @@ function makeImage(data, cb) {
...
 }
 
-module.exports = makeImage;
+function encapsulatingMakeImage(data, cb) {
+  var makeImageDomain = domain.create();
+  makeImageDomain.on('error', function(err) {
+    console.error('Badge generator error:', err.stack);
+    cb('', err);
+  });
+  makeImageDomain.bind(makeImage).apply(this, arguments);
```
Comment (espadrine) [inline badge.js]: I'd rather avoid using `arguments` if possible.

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+node_modules
+.git
```
Comment (espadrine) [inline .dockerignore]: I have to dig into what those lines do. Could you provide a pointer?

**Comment by espadrine** [comment]
Could you detail what the point of adding a domain there is?

File: .dockerignore (hunk 2 above)
Comment (ElNounch) [inline .dockerignore]: It slipped. :innocent:
Documentation [is here](https://docs.docker.com/reference/builder/#dockerignore-file), in short it speeds up docker image build by not copying the ~50M of git and node modules stuff.

**Comment by ElNounch** [comment]
I added a domain here when I was doing some hairy stuff while generating badges.
No rush to add one. Yet... :relaxed:

File: .dockerignore (hunk 2 above)
Comment (espadrine) [inline .dockerignore]: I see. So I'll have to be careful not to deploy code without package.json set up correctly.

(I'm not currently using docker, but may switch very soon, which is why these questions are on my mind ☺)

**Comment by espadrine** [comment]
Did you hit errors? Could we fix those?

**Comment by AdrieanKhisbe** [comment]
> $ touch this

**Comment by olivierlacan** [comment]
This is a stale PR. And there's no clear justification for the change. Closing.

**Pull request closed without merging**